lib/core/kotekanTrackers.cpp

```diff
@@ -0,0 +1,68 @@
+#include "kotekanTrackers.hpp"
+
+#include "kotekanLogging.hpp" // for ERROR_NON_OO
+
+#include "fmt.hpp"  // for format, fmt
+#include "json.hpp" // for json, basic_json<>::value_type
+
+#include <functional> // for _Bind_helper<>::type, _Placeholder, bind, _1, placeholders
+#include <stdexcept>  // for runtime_error
+#include <utility>    // for pair
+
+namespace kotekan {
+namespace trackers {
+
+KotekanTrackers::KotekanTrackers() {}
+
+KotekanTrackers::~KotekanTrackers() {
+    restServer::instance().remove_get_callback("/trackers");
+}
+
+KotekanTrackers& KotekanTrackers::instance() {
+    static KotekanTrackers _instance;
+    return _instance;
+}
+
+void KotekanTrackers::register_with_server(restServer* rest_server) {
+    using namespace std::placeholders;
+    rest_server->register_get_callback("/trackers",
+                                       std::bind(&KotekanTrackers::trackers_callback, this, _1));
+}
+
+void KotekanTrackers::trackers_callback(connectionInstance& conn) {
+    nlohmann::json trackers_json = {};
+
+    for (auto& it : trackers) {
+        trackers_json[it.first] = it.second->get_json();
```

Comment:
hmm, we might want to think about the structure of this a little more.  We kind of want to group these clearly by the stages `unique_name` so that we can associate them with the pipeline graph easily.